### PR TITLE
Use GitPython to crawl Looker repo [sc-5045]

### PR DESCRIPTION
**Pull request by alyiwang**
### 🤔 Why?

Provides the capability to fetch LookML project from git repo

### 🤓 What?

- use GitPython to clone LookML repo
- Update config

### 🧪 Tested?

locally run crawler with access token

### CODE REVIEW

**Comment by mars-lan** [comment]
Great work. Please also update README with updated instructions.

File: metaphor/common/git.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+def clone_repo(config: GitRepoConfig, local_dir: Optional[str] = None) -> str:
+    """
+    Clone a git repo to local storage and return the path to the project
+    If local directory not provided, use generated temp directory
+    """
+    if local_dir is None:
+        local_dir = tempfile.mkdtemp()
+
+    parsed_url = urlparse(config.git_url)
+    auth = config.access_token or f"{config.username}:{config.password}"
```
Comment (mars-lan) [inline metaphor/common/git.py:47]: I thought you still need to specify `{username}:{access_token}`? Or can `username` be omitted?

Comment (mars-lan) [reply metaphor/common/git.py:47]: For GitLab it seems that the user name should be hard-coded to `oauth2`: https://stackoverflow.com/a/67534214

Comment (alyiwang) [reply metaphor/common/git.py:47]: For github, the token alone can get access, doesn't need the username. Tested with our looker repo. I will test Gitlab URL too.

Comment (alyiwang) [reply metaphor/common/git.py]: Added support for GitLab

File: metaphor/common/git.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    Clone a git repo to local storage and return the path to the project
+    If local directory not provided, use generated temp directory
+    """
+    if local_dir is None:
+        local_dir = tempfile.mkdtemp()
+
+    parsed_url = urlparse(config.git_url)
+    assert parsed_url.netloc in _supported_git_services, "Unsupported git service"
+
+    if parsed_url.netloc == "github.com":
```
Comment (mars-lan) [inline metaphor/common/git.py:52]: This won't work for GitHub enterprise. Alternatively, we can ask the user to always specify `username` and add a note in the README that for GitLab, it should be set to `oauth2`.

File: metaphor/common/docs/git_repo.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Git Repo Connection Config
```
Comment (mars-lan) [inline metaphor/common/docs/git_repo.md:1]: Awesome. We can potentially re-use this for dbt core as well!

File: metaphor/common/git.py (hunk 2 above)
Comment (alyiwang) [reply metaphor/common/git.py]: Good point. Made `username` as required, even though for github access token, it's not needed.
Also tested against bitbucket.

File: metaphor/common/docs/git_repo.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+Using personal access token with read scope is the recommended way to connect to git repository. To create a new token, please refer to the instructions for [Github](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) or [GitLab](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html#create-a-personal-access-token). The configuration will be:
+
+```yaml
+git_url: <git_repo_url> # ending with .git
+username: <username>
+access_token: <personal_access_token>
+```
+
+**NOTE**: For GitLab or GitLab enterprise, please use `oauth2` as the username.
```
Comment (mars-lan) [inline metaphor/common/docs/git_repo.md:15]: Actually only GitLab. GitHub enterprise will be the same as GitHub except a different URL.

Comment (mars-lan) [reply metaphor/common/docs/git_repo.md:15]: Also use block quote for consistency.

**Review by mars-lan — COMMENTED** [review]
Otherwise LGTM!

File: metaphor/common/docs/git_repo.md (hunk 4 above)
Comment (alyiwang) [reply metaphor/common/docs/git_repo.md:15]: It's `GitLab` enterprise

Comment (mars-lan) [reply metaphor/common/docs/git_repo.md:15]: My bad 😄

Comment (alyiwang) [reply metaphor/common/docs/git_repo.md]: updated

Comment (mars-lan) [reply metaphor/common/docs/git_repo.md]: What I meant is block quote like this:

> Note: ...

Comment (alyiwang) [reply metaphor/common/docs/git_repo.md]: 😢  What was I thinking. Updated

**Review by mars-lan — APPROVED** [review]

File: metaphor/looker/README.md [hunk 5]
```diff
@@ -71,12 +70,28 @@ Note that `connections` is a mapping of database connection names to connection
...
+
+```yaml
+lookml_dir: <path_to_lookml_project>
+```
+
+Alternatively, if the source code is hosted on Git, we can set the git repository config as following:
+
+```yaml
+lookml_git_repo:
+  git_url: <git_repo_url>
```
Comment (mars-lan) [inline metaphor/looker/README.md:83]: Sorry missed this before the merge. Need to also add user name here

Comment (alyiwang) [reply metaphor/looker/README.md:83]: Good catch

**Pull request merged**